app/src/main/java/com/vultisig/wallet/presenter/keysign/Keysign.kt

```diff
@@ -164,10 +167,27 @@ fun PreviewKeysign() {
                 ),
                 utxos = emptyList(),
                 vaultPublicKeyECDSA = ""
-            ), gson = Gson(), thorChainApi = ThorChainApiImpl(
+            ),
+            gson = Gson(),
+            thorChainApi = ThorChainApiImpl(
                 gson = Gson(), httpClient = HttpClient()
-            )
+            ),
+            blockChairApi = object : BlockChairApi {
+                override suspend fun getAddressInfo(coin: Coin): BlockchairInfo? {
+                    return null
+                }
 
+                override suspend fun getBlockchairStats(coin: Coin): BigInteger {
+                    return BigInteger.ZERO
+                }
+
+                override suspend fun broadcastTransaction(
+                    coin: Coin,
+                    signedTransaction: String,
+                ): String {
+                    return ""
+                }
+            }
```

Comment:
You can make your ViewModel class a HiltViewModel and set it up as a default parameter for this Composable as:
`viewModel: KeysignViewModel = hiltViewModel()`

This will avoid using mocks here, and even instantiating KeysignViewModel

Reply:
a lot of information needed to create KeysignViewModel is passed down from it's parent , I am not sure how to make it using DI yet ,  
``` 
private val vault: Vault,
    private val keysignCommittee: List<String>,
    private val serverAddress: String,
    private val sessionId: String,
    private val encryptionKeyHex: String,
    private val messagesToSign: List<String>,
    private val keyType: TssKeyType,
    private val keysignPayload: KeysignPayload,
```
All these information are decided in KeysignFlowViewModel or JoinKeysignViewModel

Reply:
Good! It's probably SavedStateHandle :)